bbolt/bbolt.go

```diff
@@ -210,16 +195,36 @@ func (t bboltShelf) Get(key []byte) ([]byte, error) {
 	return append(value[:0:0], value...), nil
 }
 
-func (t bboltShelf) Put(key []byte, value []byte) error {
-	return t.bucket.Put(key, value)
+func (t bboltShelf) Put(key stoabs.Key, value []byte) error {
+	return t.bucket.Put(key.Bytes(), value)
 }
 
-func (t bboltShelf) Delete(key []byte) error {
-	return t.bucket.Delete(key)
+func (t bboltShelf) Delete(key stoabs.Key) error {
+	return t.bucket.Delete(key.Bytes())
 }
 
 func (t bboltShelf) Stats() stoabs.ShelfStats {
 	return stoabs.ShelfStats{
 		NumEntries: uint(t.bucket.Stats().KeyN),
 	}
 }
+
+func (t bboltShelf) Iterate(callback stoabs.CallerFn) error {
+	cursor := t.bucket.Cursor()
+	for k, v := cursor.First(); k != nil; k, v = cursor.Next() {
+		if err := callback(stoabs.BytesKey(k), v); err != nil {
```

Comment:
in `Get()` we clone the value we return to avoid the BBolt pitfall of using it outside of the TX (resulting in random memory being returned). Maybe do that here as well.